### PR TITLE
[#10955] Remove all references to internal Datastore API usage

### LINKED ISSUE #10955: Migrate Objectify 5 to Objectify 6
This issue keeps track of PRs developing features in the title.

**Environment**: 
All PRs for this feature should be made against `objectify-v6-migration` branch **[⚠️ IMPORTANT]**

**Description of feature/enhancement:**
See https://github.com/TEAMMATES/teammates/wiki/Google-App-Engine-Runtime-Upgrade for project details.

**Related past issues:**
#9131

Epic will be resolved after merging of feature branch to master.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #10955 

All remaining references to internal Datastore API are removed, and Remote API is replaced with direct connection to Google Cloud Datastore instance.

### CODE REVIEW

File: src/main/java/teammates/storage/api/OfyHelper.java [hunk 1]
```diff
@@ -24,6 +24,14 @@
  **/
 public class OfyHelper implements ServletContextListener {
 
+    private static void initializeDatastore() {
+        DatastoreOptions.Builder builder = DatastoreOptions.newBuilder().setProjectId(Config.APP_ID);
+        if (Config.isDevServer()) {
+            builder.setHost("http://localhost:" + Config.APP_LOCALDATASTORE_PORT);
+        }
+        ObjectifyService.init(new ObjectifyFactory(builder.build().getService()));
+    }
```
Comment (Derek-Hardy) [inline src/main/java/teammates/storage/api/OfyHelper.java:33]: 👍 Great refactoring!

**Review by Derek-Hardy — APPROVED** [review]
LGTM!

**Comment by jianhandev** [comment]
LGTM!

**Review by jianhandev — APPROVED** [review]
LGTM!

**Pull request merged**